src/polyfem/io/OutData.cpp

```diff
@@ -1228,7 +1184,12 @@ namespace polyfem::io
 			}
 		}
 
-		VTUWriter writer;
+		std::shared_ptr<paraviewo::ParaviewWriter> tmpw;
+		if (opts.use_hdf5)
+			tmpw = std::make_shared<paraviewo::HDF5VTUWriter>();
+		else
+			tmpw = std::make_shared<paraviewo::VTUWriter>();
+		paraviewo::ParaviewWriter &writer = *tmpw;
```

Comment:
Are you sure tmpw wont be garbage collected leaving writer invalid?

Reply:
Is writer just a local variable?

Reply:
it is, but I want it polymorphic. Is there a better way?

Reply:
Just use the shared  pointer?